sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -239,25 +239,29 @@ public ConfigurationClientBuilder endpoint(String endpoint) {
      * @param connectionString Connection string in the format "endpoint={endpoint_value};id={id_value};
      * secret={secret_value}"
      * @return The updated ConfigurationClientBuilder object.
-     * @throws NullPointerException If {@code credential} is {@code null}.
+     * @throws NullPointerException If {@code connectionString} is {@code null}.
+     * @throws IllegalArgumentException if {@code connectionString} is an empty string. Or the secret is invalid and
+     * cannot instantiate HMAC-SHA256 MAC algorithm.
      */
     public ConfigurationClientBuilder connectionString(String connectionString) {
-        Objects.requireNonNull(connectionString);
+        Objects.requireNonNull(connectionString, "'connectionString' cannot be null.");
+
+        if (CoreUtils.isNullOrEmpty(connectionString)) {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("'connectionString' cannot be an empty string."));
+        }
 
         try {
             this.credential = new ConfigurationClientCredentials(connectionString);
         } catch (InvalidKeyException err) {
             throw logger.logExceptionAsError(new IllegalArgumentException(
-                    "The secret is invalid and cannot instantiate the HMAC-SHA256 algorithm.", err));
+                "The secret is invalid and cannot instantiate the HMAC-SHA256 algorithm.", err));
```

Comment:
Should we add in that the secret contained within the connection string is invalid? Right now this is too vague and being thrown from a black box scenario for anyone consuming this SDK.